src/asm2wasm.h

```diff
@@ -1440,7 +1162,7 @@ void Asm2WasmBuilder::processAsm(Ref ast) {
         if (importResult == f64) {
           // we use a JS f64 value which is the most general, and convert to it
           switch (old) {
-            case i32: replaceCurrent(parent->makeTrappingFloatToInt32(true /* signed, asm.js ffi */, curr)); break;
+            case i32: replaceCurrent(parent->builder.makeUnary(TruncSFloat64ToInt32, curr)); break;
```

Comment:
didn't we say we'd avoid using the pass in asm2wasm, and have it call directly to utility methods to do this? to avoid all the issues with having a pass be self-contained, and run at the right time, and all that?

maybe I misuderstood what we agreed on.

Reply:
Maybe I did. I thought we only cared about doing that in order to avoid potentially optimizing trapping opts into unconditional wasm traps.
As far as correctness goes, if that's not a concern then this is still fine.
This does change the visibility of the clamping to the optimizer, which I was assuming wasn't as important because of the inherent slowness of clamp/js modes as it was.

If we do care, I still think we should use a pass, with the following approach:
1. change the pass to be function-parallel
2. run it as early as possible (right after AutoDrop)
2. generate all the generated functions all the time, because we don't know which ones we need
3. let DCE remove the ones we don't need

Reply:
I think we should care, yeah: if the optimizer works on incorrect code (before clamping was applied) then it doesn't see the right side effects, might precompute incorrectly, etc.

The plan 1-4 sounds ok. Hopefully with that the test outputs will be unchanged. (The only change I'd expect to see is that the order of functions might differ, since we add the helper functions themselves at a different time, that's ok of course.)